plutus-contract/src/Language/Plutus/Contract/Trace/RequestHandler.hs

```diff
@@ -52,3 +81,82 @@ wrapHandler (RequestHandler h) = RequestHandler $ \Request{rqID, itID, rqRequest
 
 maybeToHandler :: (req -> Maybe resp) -> RequestHandler effs req resp
 maybeToHandler f = RequestHandler $ maybe empty pure . f
+
+-- handlers for common requests
+
+handleOwnPubKey ::
+    forall a effs.
+    ( Member WalletEffect effs
+    , Member Log effs
+    )
+    => RequestHandler effs a PubKey
+handleOwnPubKey =
+    RequestHandler $ \_ ->
+        surroundDebug "handleOwnPubKey" Wallet.Effects.ownPubKey
+
+handleSlotNotifications ::
+    forall effs.
+    ( Member WalletEffect effs
+    , Member Log effs
+    )
+    => RequestHandler effs Slot Slot
+handleSlotNotifications =
+    RequestHandler $ \targetSlot ->
+        surroundDebug "handleSlotNotifications" $ do
+            currentSlot <- Wallet.Effects.walletSlot
+            logDebug $ Text.pack $ "targetSlot: " <> show targetSlot <> "; current slot: " <> show currentSlot
+            guard (currentSlot >= targetSlot)
+            pure currentSlot
+
+handlePendingTransactions ::
+    forall effs.
+    ( Member WalletEffect effs
+    , Member Log effs
+    , Member SigningProcessEffect effs
+    , Member ChainIndexEffect effs
+    )
+    => RequestHandler effs UnbalancedTx (Either WalletAPIError Tx)
+handlePendingTransactions =
+    RequestHandler $ \unbalancedTx ->
+        surroundDebug "handlePendingTransactions" $ do
+        logDebug "Start watching contract addresses."
+        wa <- Wallet.Effects.watchedAddresses
+        traverse_ Wallet.Effects.startWatching (AM.addressesTouched wa (unBalancedTxTx unbalancedTx))
+        (Right <$> Wallet.handleTx unbalancedTx) `Eff.handleError` (\err -> logWarn "handleTxFailed" >> pure (Left err))
+
+handleUtxoQueries ::
+    forall effs.
+    ( Member Log effs
+    , Member ChainIndexEffect effs
+    )
+    => RequestHandler effs Address UtxoAtAddress
+handleUtxoQueries = RequestHandler $ \addr ->
+    surroundDebug "handleUtxoQueries" $ do
+        Wallet.Effects.startWatching addr
+        AddressMap utxoSet <- Wallet.Effects.watchedAddresses
+        maybe empty (pure . UtxoAtAddress addr) $ Map.lookup addr utxoSet
```

Comment:
I still find `maybe` a bit hard to read because I forget which way the arguments go. If you're not using it in a pipeline I think pattern matching is nicer.